### PR TITLE
Render dropped events in background cells

### PR DESCRIPTION
**Pull request by tigleym**
Hello, I've been following your PR and wanted to thank you for all the work you've put into it! It would be really great to see your changes merged in sometime in the future. 

I've been playing around with your dnd-example branch and started extending upon it, particularly rendering dropped events within their new background cells. It is still a work in progress as I need to correctly update new event times/days upon drop, but other than that there is not much left for me to expand on. 
If you had any questions, or suggestions for this PR just let me know!

### CODE REVIEW

**Comment by srajko** [comment]
@TigleyM thank you so much for submitting this!  I will try to review it soon.

**Comment by srajko** [comment]
Good call on breaking up the example into separate files.

**Comment by srajko** [comment]
This is a clever use of react context - it gets around the problem that the composition path from the top-level calendar to the nested drag/drop components isn't entirely under the example's control.

File: examples/demos/dnd.js [hunk 1]
```diff
@@ -1,98 +0,0 @@
...
-    propsCopy.components = merge({}, this.props.components, {
-      eventWrapper: DraggableEventWrapper,
-      backgroundWrapper: DroppableBackgroundWrapper
-    })
-    return <BigCalendar {...propsCopy} />
-  }
-}
-
-
-let Dnd = React.createClass({
```
Comment (srajko) [inline examples/demos/dnd.js:87]: The example was originally split into the `DndBigCalendar` component and the `Dnd` export, to show that you could have a general `DndBigCalendar` that isn't tied to a specific set of events or dnd update behavior.  That way we could extract `DndBigCalendar` into a separate module (react-big-calendar addon) down the road.

Could you restore that structure in your code?  I would say that `moveEvent`, `defaultDate`, and `events` should be defined in the outer component, though you could perhaps have `DndBigCalendar` receive `moveEvent` as a prop (so `moveEvent` is implemented in `Dnd`, then passed to `DndBigCalendar` as a prop, and `DndBigCalendar` makes it available as part of its context).

With your nice separation of files, you could also have `DndBigCalendar` in a separate file - that would isolate the general component code from the code that uses it.

File: examples/demos/dnd/DroppableBackgroundWrapper.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+/* drop targets */
+const dropTarget = {
+  drop(props, monitor, backgroundWrapper) {
+    const event = monitor.getItem();
+    const { moveEvent } = backgroundWrapper.context
+    const updatedEvent = updateEventTime(event, backgroundWrapper.props)
+    moveEvent(updatedEvent)
+    alert(`${event.title} was dropped onto ${backgroundWrapper.props.type} ${backgroundWrapper.props.value}`);
```
Comment (srajko) [inline examples/demos/dnd/DroppableBackgroundWrapper.js]: Now that you have `moveEvent`, the alert should probably go in there.

File: examples/demos/dnd/dropActions.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  const newStartDate = `${moment(value).format('MM/DD/YYYY')} ${type === 'Day' ? moment(start).format('HH:mm:ss') : newStartTimeSlot}`
+  const newEndDate = `${moment(value).add(dayDifference, 'days').format('MM/DD/YYYY')} ${type === 'Day' ? moment(end).format('HH:mm:ss') : newEndTimeSlot}`
+
+  const newEventTimes = {
+    ...event,
+    start: moment(newStartDate).toDate(),
+    end: moment(newEndDate).toDate()
+  }
+
+  return newEventTimes
```
Comment (srajko) [inline examples/demos/dnd/dropActions.js]: Would it be possible to perform the `start` and `end` calculations directly through the `moment`s, without converting to strings and parsing them?

**Comment by tigleym** [comment]
Thank you for taking the time to leave some notes with your review! I've restored the original component structure in regards to splitting `DndBigCalendar` and `Dnd` . As soon as I have time I will take a closer look at Moment JS's docs for calculating `start` and `end`.

**Comment by srajko** [comment]
@TigleyM sorry for the delay, I was out of town for a few days.

This looks great to me - was there anything else you were thinking about adding?  If not, I'd say just squash this into a single commit to clean up the history, and I'll merge it in.

Thanks again for the contribution, I think it will make https://github.com/intljusticemission/react-big-calendar/pull/84 much better.

**Comment by tigleym** [comment]
@srajko  Cool, I've cleaned up the commits so whenever you're ready you can merge the PR in. 

I've been playing around with implementing resizeable events on a separate branch, but I'll keep that as a separate PR if I ever manage to get a good working example. Other than that, I think everything is good to go. Let me know if there is anything else you'd like to me to fix/polish though!

**Pull request merged**